include/envoy/secret/secret_manager.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/secret/secret.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * A manager for all static secrets.
+ */
+class SecretManager {
+public:
+  virtual ~SecretManager() {}
+
+  /**
+   * @param secret Updated Secret.
+   */
+
+  virtual void addOrUpdateStaticSecret(const SecretSharedPtr secret) PURE;
+
+  /**
+   * @return the static secret for the given name.
+   */
+  virtual const SecretSharedPtr staticSecret(const std::string& name) const PURE;
```

Comment:
Do we care whether the secret was configured as static or not? I think the only difference between "static" and "dynamic" secrets is going to be the missing config source, so perhaps something like that:

```
const SecretSharedPtr findSecret(const std::string& name, const std::string &source);
```

could be used for both, with `source` being `EMPTY_STRING` for the static secrets?

Reply:
renamed the method. 
Since source paramter is not going to be used for now, I will add source parameter when it supports dynamic secret.